liquidprompt

```diff
@@ -948,14 +951,34 @@ _lp_git_branch_color()
                 d_lines=0
             fi
 
-            local has_lines
-            has_lines="+$i_lines/-$d_lines"
+            diff_details+="+$i_lines/-$d_lines"
+        fi
+
+        local status_details=""
+        if (( LP_ENABLE_GIT_STATUS )); then
+            local status_lines="$(LC_ALL=C \git status --porcelain 2>/dev/null)"
+
+            if [[ -n "$status_lines" ]]; then
+                local staged="$(echo "${status_lines}" | \grep '^[^? ]. .*' | wc -l)"
+                local unstaged="$(echo "${status_lines}" | \grep '^.[^? ] .*' | wc -l)"
+                local untracked="$(echo "${status_lines}" | \grep '^?? .*' | wc -l)"
+
+                [[ "$staged" -gt 0 ]] && status_details+=" $LP_MARK_STAGED$staged"
```

Comment:
Not sure why we need to spit out the values that are empty, I did this:

```
if (( staged > 0 )); then
                  [[ "$staged" -gt 0 ]] && status_details+=" $LP_MARK_STAGED$staged"
                fi
                if (( unstaged > 0 )); then
                  [[ "$unstaged" -gt 0 ]] && status_details+=" $LP_MARK_UNSTAGED$unstaged"
                fi
                if (( untracked > 0 )); then
                  [[ "$untracked" -gt 0 ]] && status_details+=" ?$untracked"
                fi
```

I'm not entirely sure what the benefit of creating a single variable is as per @Rycieos but am not against it either.

Reply:
Why should we check twice? :confused: 
Isn't `if (( untracked > 0 )); then` and `[[ "$untracked" -gt 0 ]] &&` are the same thing?

Reply:
@MunifTanjim I didn't know :) Not too good with bash

Reply:
@KINKCreative If there's no more problem, maybe you can approve the changes?